### PR TITLE
10 datahike release

### PR DESCRIPTION
**Pull request by kordano**

### CODE REVIEW

File: _posts/2019-09-17-datahike-release-0.2.0.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
+---
+layout: post
+title: Datahike Release 0.2.0
+introduction: Latest datahike version is released supporting schema flexibility and time travel
+author: Konrad Kühne
+user: konrad
+refs: https://github.com/replikativ/datahike, https://github.com/lambdaforge/datalog-parser, https://www.jobtechdev.se, https://datopia.io, https://github.com/sixthnormal/clj-3df, https://docs.datomic.com, https://github.com/tonsky/datascript
+---
+
+Last week I released version `0.2.0` of **datahike** which sees new features
```
Comment (whilo) [inline _posts/2019-09-17-datahike-release-0.2.0.md]: sees -> brings (that gives the reader the direct impression that they get something)

File: _posts/2019-09-17-datahike-release-0.2.0.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+datalog parsing functionality we moved all functions and tests around datalog
+query parsing into a [separate
+project](https://github.com/lambdaforge/datalog-parser) that will also support
+functions not used in **datahike**, like creating datalog queries from its syntax
+tree, so you can parse a datalog query, add addtionaly data, and create an
+optimized query for a database to use.
+
+# Schema Flexibility
+
+Up until now **datahike** did not enforce any schema on its data, so any arbitrary
```
Comment (whilo) [inline _posts/2019-09-17-datahike-release-0.2.0.md]: any arbitrary -> badly shaped data could be added accidentally. (the rest of the sentence I would skip, it is a technical detail of datascript). We should say that schema-on-read is still supported.

File: _posts/2019-09-17-datahike-release-0.2.0.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+schema additionally as transactions in
+the index, which allows queries and time travel
+that can help auditing schema changes.
+
+# Time Travel
+
+Modern database solutions require auditable data history without integrating time stamp
+attributes to all data models. Similar to [Datomic' historic data](https://docs.datomic.com/on-prem/getting-started/see-historic-data.html)
+**datahike** supports time travel capabilities that allows datalog queries against
+the whole history, against a view at a certain point in time, or against the
```
Comment (whilo) [inline _posts/2019-09-17-datahike-release-0.2.0.md]: Two times "a certain point in time". I would say "against any point in the past".

File: _posts/2019-09-17-datahike-release-0.2.0.md [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+difference since a certain point in time.
+
+In accordance to the [GDPR regulations](https://gdpr-info.eu/) data
+purging capabilities were added that completely removes data either from
```
Comment (whilo) [inline _posts/2019-09-17-datahike-release-0.2.0.md]: removes -> remove

File: _posts/2019-09-17-datahike-release-0.2.0.md [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+
+With addition of the latest features, **datahike** required better configuration.
+Therefore a clean concept was implemented that allowed picking only these
+functions from datahike that you need the most at database creation. For example
+both time travel and schema-on-read capabilities can be deactivated if you have
+no use for it by setting the according toggle.
+
+# Examples
+
+Since it is always helpful to have examples I created [projects](https://github.com/replikativ/datahike/tree/master/examples) inside
```
Comment (whilo) [inline _posts/2019-09-17-datahike-release-0.2.0.md]: I created -> we provide (the person talking is not clear from context)

File: _posts/2019-09-17-datahike-release-0.2.0.md [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+the code base that show different features as well as documentation about
+new features like [time
+travel](https://github.com/replikativ/datahike/blob/master/doc/time_variance.md),
+[schema](https://github.com/replikativ/datahike/blob/master/doc/schema.md), and [configuration](https://github.com/replikativ/datahike/blob/master/doc/config.md). These projects just need a REPL
+and can be worked through at your own pace. Anywhere possible I added
+explanations and different ways to use a certain feature. Have a look at the
+[basic
+example](https://github.com/replikativ/datahike/tree/master/examples/basic) for
+a first look at **datahike** showing off topics like stores, schema, or time
```
Comment (whilo) [inline _posts/2019-09-17-datahike-release-0.2.0.md]: "have a look for a first look" is not good

File: _posts/2019-09-17-datahike-release-0.2.0.md [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+
+In the next weeks I will add more examples about queries and transactions.
+
+# Future Plans
+
+For the next release we want to make datalog available to a wider audience, so
+we will will release API bindings for Java/Scala of **datahike**.
+
+In collaboration with our partners from the [3DF
+project](https://github.com/sixthnormal/clj-3df) we want to support remote capabilities with a HTTP layer, having only a thin client
```
Comment (whilo) [inline _posts/2019-09-17-datahike-release-0.2.0.md]: we want -> we aim for (wanting is  a bit strong in English and can sound a bit desparate, it is better to say "plan to", "work on" and "aim for" or similar phrases).

File: _posts/2019-09-17-datahike-release-0.2.0.md [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+on the application side instead of the full **datahike** stack.
+
+With support of a datalog query planner we want to be able to predict the
+costs of each query and recommend possible solutions to improve your overall queries.
+
+For security reasons it would be useful to have an identity and access
```
Comment (whilo) [inline _posts/2019-09-17-datahike-release-0.2.0.md]: This sentence is a bit lonely. Maybe add a link to Wikipedia or IAM reference to show that we have a concept that people can discuss with us.

File: _posts/2019-09-17-datahike-release-0.2.0.md [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+management concept ready to be used.
+
+Since we have experience with probabilistic programming in
```
Comment (whilo) [inline _posts/2019-09-17-datahike-release-0.2.0.md]: Nice! :)

File: _posts/2019-09-17-datahike-release-0.2.0.md [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+Last week I released version `0.2.0` of **datahike** which sees new features
+and marks a step away from it's datascript origin. The mayor new features are
+schema flexibility and time travel capabilities. While integrating the
+latest datascript code we extended it through protocols on storage and index level.
+
+# Datascript Integration and Cleanup
+
+While **datahike** began as a fork of
+[datascript](https://github.com/tonsky/datascript) and integration of the
+hitchhiker tree, we are now adding features of our own and moving away from the
```
Comment (whilo) [inline _posts/2019-09-17-datahike-release-0.2.0.md]: adding features of our own -> adding novel features. That they are our features is not as important to the reader and sounds a bit like as if we are sorry that it previously everything was just copied.

File: _posts/2019-09-17-datahike-release-0.2.0.md [hunk 11]
```diff
@@ -0,0 +1,133 @@
...
+
+# Time Travel
+
+Modern database solutions require auditable data history without integrating time stamp
+attributes to all data models. Similar to [Datomic' historic data](https://docs.datomic.com/on-prem/getting-started/see-historic-data.html)
+**datahike** supports time travel capabilities that allows datalog queries against
+the whole history, against a view at a certain point in time, or against the
+difference since a certain point in time.
+
+In accordance to the [GDPR regulations](https://gdpr-info.eu/) data
```
Comment (whilo) [inline _posts/2019-09-17-datahike-release-0.2.0.md]: This is super cool!

File: _posts/2019-09-17-datahike-release-0.2.0.md (hunk 5 above)
Comment (whilo) [reply _posts/2019-09-17-datahike-release-0.2.0.md]: (I is generally bad in these documents or scientific documents)

File: _posts/2019-09-17-datahike-release-0.2.0.md [hunk 12]
```diff
@@ -0,0 +1,133 @@
...
+# Future Plans
+
+For the next release we want to make datalog available to a wider audience, so
+we will will release API bindings for Java/Scala of **datahike**.
+
+In collaboration with our partners from the [3DF
+project](https://github.com/sixthnormal/clj-3df) we want to support remote capabilities with a HTTP layer, having only a thin client
+on the application side instead of the full **datahike** stack.
+
+With support of a datalog query planner we want to be able to predict the
```
Comment (whilo) [inline _posts/2019-09-17-datahike-release-0.2.0.md]: same with "want"

File: _posts/2019-09-17-datahike-release-0.2.0.md (hunk 9 above)
Comment (whilo) [reply _posts/2019-09-17-datahike-release-0.2.0.md]: Let me repair the current anglican website today and we can use http://anglican.ml again.

File: _posts/2019-09-18-datahike-release-0.2.0.md [hunk 13]
```diff
@@ -0,0 +1,137 @@
...
+layout: post
+title: Datahike Release 0.2.0
+introduction: Latest datahike version is released supporting schema flexibility and time travel
+author: Konrad Kühne
+user: konrad
+refs: https://github.com/replikativ/datahike, https://github.com/lambdaforge/datalog-parser, https://www.jobtechdev.se, https://datopia.io, https://github.com/sixthnormal/clj-3df, https://docs.datomic.com, https://github.com/tonsky/datascript
+---
+
+Last week we released version `0.2.0` of **datahike** which brings new features
+and marks a step away from its datascript origin. The mayor new features are
```
Comment (jsmassa) [inline _posts/2019-09-18-datahike-release-0.2.0.md]: mayor -> major

File: _posts/2019-09-18-datahike-release-0.2.0.md [hunk 14]
```diff
@@ -0,0 +1,137 @@
...
+
+In contrast to datascript's schema in the database record, **datahike** stores the
+schema additionally as transactions in
+the index, which allows queries and time travel
+that can help auditing schema changes.
+
+# Time Travel
+
+Modern database solutions require auditable data history without integrating time stamp
+attributes to all data models. Similar to [Datomic' historic data](https://docs.datomic.com/on-prem/getting-started/see-historic-data.html)
```
Comment (jsmassa) [inline _posts/2019-09-18-datahike-release-0.2.0.md]: Datomic's?

File: _posts/2019-09-18-datahike-release-0.2.0.md [hunk 15]
```diff
@@ -0,0 +1,137 @@
...
+all indices or only from the historical indices.
+
+In order to keep the current data view fast and clean, a separate set of
+indices were created that track all past data. This separation makes it very easy
+to clean up data after certain retention periods without interfering with the
+current view of data.
+
+# Configuration
+
+With addition of the latest features, **datahike** required better configuration.
```
Comment (jsmassa) [inline _posts/2019-09-18-datahike-release-0.2.0.md]: With _the_ addition

File: _posts/2019-09-18-datahike-release-0.2.0.md [hunk 16]
```diff
@@ -0,0 +1,137 @@
...
+Therefore a clean concept was implemented that allowed picking only these
```
Comment (jsmassa) [inline _posts/2019-09-18-datahike-release-0.2.0.md]: allowed -> allows or "to allow"
these -> those/ the

File: _posts/2019-09-18-datahike-release-0.2.0.md [hunk 17]
```diff
@@ -0,0 +1,137 @@
...
+functions from datahike that you need the most at database creation. For example
```
Comment (jsmassa) [inline _posts/2019-09-18-datahike-release-0.2.0.md]: For example**,** ...

File: _posts/2019-09-18-datahike-release-0.2.0.md [hunk 18]
```diff
@@ -0,0 +1,137 @@
...
+both time travel and schema-on-write capabilities can be deactivated if you have
+no use for it by setting the according toggle.
```
Comment (jsmassa) [inline _posts/2019-09-18-datahike-release-0.2.0.md]: ... _by setting the according toggle_.  -> Is that important how it is done? Too many sub-clauses...

File: _posts/2019-09-18-datahike-release-0.2.0.md [hunk 19]
```diff
@@ -0,0 +1,137 @@
...
+
+# Examples
+
+Since it is always helpful to have examples we provide [projects](https://github.com/replikativ/datahike/tree/master/examples) inside
```
Comment (jsmassa) [inline _posts/2019-09-18-datahike-release-0.2.0.md]: have examples**,** we provide

File: _posts/2019-09-18-datahike-release-0.2.0.md [hunk 20]
```diff
@@ -0,0 +1,137 @@
...
+the code base that show different features as well as documentation about
+new features like [time
+travel](https://github.com/replikativ/datahike/blob/master/doc/time_variance.md),
+[schema](https://github.com/replikativ/datahike/blob/master/doc/schema.md), and
+[configuration](https://github.com/replikativ/datahike/blob/master/doc/config.md). 
+
+These projects just need a REPL and can be worked through at your own pace.
+Anywhere possible we added explanations and different ways to use a certain
```
Comment (jsmassa) [inline _posts/2019-09-18-datahike-release-0.2.0.md]: -> Where it was possible... (maybe)

File: _posts/2019-09-18-datahike-release-0.2.0.md [hunk 21]
```diff
@@ -0,0 +1,137 @@
...
+feature. Have a look at the [basic
+example](https://github.com/replikativ/datahike/tree/master/examples/basic)
+which introduces topics like stores, schema or time travel.
+Over the next weeks we will add more examples about queries and transactions.
+
+# Future Plans
+
+For the next release we try to make datalog available to a wider audience by
```
Comment (jsmassa) [inline _posts/2019-09-18-datahike-release-0.2.0.md]: we try -> we are planning to

File: _posts/2019-09-18-datahike-release-0.2.0.md [hunk 22]
```diff
@@ -0,0 +1,138 @@
...
+In contrast to datascript's schema in the database record, **datahike** stores the
+schema additionally as transactions in
+the index, which allows queries and time travel
+that can help auditing schema changes.
+
+# Time Travel
+
+Modern database solutions require auditable data history without integrating time stamp
+attributes to all data models. Similar to [Datomic's historic data](https://docs.datomic.com/on-prem/getting-started/see-historic-data.html)
+**datahike** supports time travel capabilities that allows datalog queries against
```
Comment (whilo) [inline _posts/2019-09-18-datahike-release-0.2.0.md]: allows -> allow

**Pull request merged**